### ISSUE TITLE
Fee calculation

### ISSUE THREAD

**Issue by Stiekemeheld** [nutchanon-ust/anchor-guard-ui#1]
Sometimes transactions fail because of gas issues, didn't look at code but maybe should be calculated dynamically![Screenshot_2022-01-24-14-01-57-30_948cd9899890cbd5c2798760b2b95377.jpg](https://user-images.githubusercontent.com/42800630/150736536-0d816bf9-5ed9-4367-b8f4-b491292c1a8c.jpg)![Screenshot_2022-01-24-14-01-53-30_948cd9899890cbd5c2798760b2b95377.jpg](https://user-images.githubusercontent.com/42800630/150736553-e441a5e6-2208-445b-8630-6c9d1ce12685.jpg)

**Comment by nutchanon-ust** [comment]
Thanks for the report. I calculated the gas fee dynamically. Currently it is using UST as gas only.
Did your wallet have enough UST for the fee? From the error, it seems to need 0.033047 UST.

**Comment by Stiekemeheld** [comment]
Yeah there is definitely enough ust left in the wallet. It happened with 2 out of the 3 wallets i tried it with, when looking at the logs it looks like it needs 1uusd more than provided. Maybe an option to hardcode that extra?

**Comment by nutchanon-ust** [comment]
I have added a gas buffer which is currently set to `0.003` UST. Please let me know if you still have this issue.
In the future releases, the gas estimation could be improved.
